### DISCUSSION TITLE
[Discussion] Using "master" as branch name

### DISCUSSION THREAD

**Discussion by pieterhijma** [FreeCAD/FreeCAD#10226] [Discussions that will turn in to issues but need refining]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Full version info

```shell
OS: Arch Linux
Word size of FreeCAD: 64-bit
Version: 0.21.0.33675 (Git)
Build type: Release
Branch: makepkg
Hash: 41b058e2087de60dc8fef3d3e68c7d0129e13abf
Python 3.11.3, Qt 5.15.10, Coin 4.0.1, Vtk 9.2.6, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * freecad.gears 1.0.0
  * Ondsel-Lens
  * BIM 2021.12.0
  * lattice2 1.0.0
  * LCInterlocking
  * parts_library
  * fasteners 0.4.56
  * Assembly4 0.50.2
```

### Subproject(s) affected?

None

### Problem description

Many communities are moving away from terms such as "master". This is also true in Git, for example discussed by the [Software Freedom Conservancy](https://sfconservancy.org/news/2020/jun/23/gitbranchname/) and [GitHub](https://github.com/github/renaming). I think it would be good if the FreeCAD project makes the same change by renaming `master` to `main`, which appears to be the most popular replacement.

It's probably best to set a date, announce it widely and provide some instructions for those who have clones on their local machines and in their GitHub account.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Pesc0** [comment]
#6807

**Comment by luzpaz** [comment]
This will be a huge headache and will break a lot of links. This was discussed in the past and the enormity of it was daunting along with the repercussions (downstream packagers, old bugs, documentation, online links, etc..etc..). I understand the sentiment but yikes...

**Reply by chennes** [reply]
GitHub will automatically retarget  PRs, and we maintain the dev build snap, appimage, and Conda builds in-house. We would not be the first project to do this (Blender, to name just a small example 😀 ), and it will only affect dev builds: if we are going to do it, now is the time. We have at least 6 months before another major release.

**Reply by luzpaz** [reply]
Alrighty then. Full speed ahead!

**Comment by chennes** [comment]
I believe from a technical standpoint only admins can make the change, so it is really that team that must make the decision. That is @wwmayer , @yorikvanhavre , @sliptonic , and @kkremitzki . I support the change, but can’t actually make it.

**Comment by yorikvanhavre** [comment]
i agree with the change too

**Reply by sliptonic** [reply]
I couldn't care less either way.  If it matters to somebody, go for it

**Comment by mfro0** [comment]
as long as there is chess masters and university master classes, I think there's nothing wrong with a master branch

**Comment by yorikvanhavre** [comment]
In any case, it seems like there is a large consensus to do it, so let's do it. @FreeCAD/maintainers , if you see any issue with this, please speak now ;)

**Reply by Pesc0** [reply]
Friendly reminder of #6807
Also please post instructions so everyone knows what to do after the change, thanks!

**Reply by yorikvanhavre** [reply]
Oh good reminder, thanks!

**Reply by luzpaz** [reply]
Lets do it. Full speed ahead!

**Comment by raharth** [comment]
I'd be really careful with that kind of change, it can cause severe issues. This change is technically unnecessary and can cause many hours of work. One of your suppliers made that change and their service was entirely unavailable for a while.

**Reply by yorikvanhavre** [reply]
Of course, always expect the worse :sweat_smile: but I think more and more of the world will begin to expect a "main" branch instead of "master", so sooner or later it will be needed probably, so better be sooner than later.

Also having this repo out of order for a little while is probably not a huge problem. There are many mirrors.

**Reply by raharth** [reply]
@yorikvanhavre Sure I see your point. At work we decided to keep previously existing ones at the old naming convention, while new ones start with a "main" branch instead. I mean you are the maintainer and I'mcertainly not trying to tell you what to do, I just wanted to raise a warning that it can caus a lot of disruption - which I personally was not expecting at all first time someone told me about changing!

**Comment by xtemp09** [comment]
@yorikvanhavre, would it be possible to just create a mirror branch, instead of renaming? In this case, all existing scripts work and you always can claim you are 100% woke.

**Comment by yorikvanhavre** [comment]
I am 100% woke already :laughing: 

I think leaving a master branch in place is dangerous because people will still rely on it although it won't be updated anymore. Better remove it so an error occurs on people's git clients and they are forced to update.

I think before doing this we need to:

* Document what people need to do to fix. Github says this:
```bash
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```
* Announce the move well in advance (1 week or so).

Let's program it for the weekend of **october 7/8th**?

BTW I'm just testing the above workflow on a dummytest repo, when the master branch is gone on github, when trying to pull from your local client, you get:
 ```
Unpacking objects: 100% (3/3), 674 bytes | 674.00 KiB/s, done.
From github.com:yorikvanhavre/testmain
 * [new branch]      main       -> origin/main
Your configuration specifies to merge with the ref 'refs/heads/master'
from the remote, but no such ref was fetched.
```
So it's not very breaking, which is good or bad, nothing will break on user's side, they just won't be able to update until they rename, but some people might not see the message.

Anyway, I think this is the way to go. Where should we publish this? Forum + github maintainers certainly, but is it worth a blog post too?

**Comment by yorikvanhavre** [comment]
Ok programmed for **sunday 8th, at 20h00 UTC**

https://forum.freecad.org/viewtopic.php?p=710744#p710744
https://blog.freecad.org/2023/10/02/renaming-development-git-branch-from-master-to-main/

**Comment by yorikvanhavre** [comment]
ok it's done! @FreeCAD/maintainers please update your local branch!
```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

**Comment by luzpaz** [comment]
Closing.
